jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/coursepublish/CourseSiteRemovalJob.java

```diff
@@ -0,0 +1,263 @@
+         if (userId == null || userId.trim().length() == 0) {
+            logger.warn("The property " + PROPERTY_COURSE_SITE_REMOVAL_USER + " was not specified in sakai.properties.  Using a default value of " + DEFAULT_VALUE_COURSE_SITE_REMOVAL_USER + ".");
+            userId = DEFAULT_VALUE_COURSE_SITE_REMOVAL_USER;
+         }
+         user = userDirectoryService.getUser(userId);
+      } catch (UserNotDefinedException ex) {
+         user = null;
+         logger.error("The user with eid " + userId + " was not found.  The course site removal job has been aborted.");
+      }
+
+      // get the action to be taken when a course is found to be expired
+      String actionString = serverConfigurationService.getString(PROPERTY_COURSE_SITE_REMOVAL_ACTION);
+      if (actionString == null || actionString.trim().length() == 0)
+      {
+         logger.warn("The property " + PROPERTY_COURSE_SITE_REMOVAL_ACTION + " was not specified in sakai.properties.  Using a default value of " + DEFAULT_VALUE_COURSE_SITE_REMOVAL_ACTION + ".");
+         action = DEFAULT_VALUE_COURSE_SITE_REMOVAL_ACTION;
+      }
+      else
+      {
+         action = getAction(actionString);
+         if (action == null) {
+            logger.error("The value specified for " + actionString + " in sakai.properties, " + PROPERTY_COURSE_SITE_REMOVAL_ACTION + ", is not valid.  A default value of " + DEFAULT_VALUE_COURSE_SITE_REMOVAL_ACTION + " will be used instead.");
+            action = DEFAULT_VALUE_COURSE_SITE_REMOVAL_ACTION;
+         }
+      }
+	}
+
+   /**
+    * implement the quartz job interface, which is called by the scheduler when a trigger associated with the job fires.
+    * this quartz job removes course sites that are more than a specified number of terms old.
+    */
+   public void execute(JobExecutionContext context) throws JobExecutionException {
+      synchronized (this) {
+         logger.info("execute()");
+
+         if (user == null) {
+            logger.error("The scheduled job to remove course sites can not be run with an invalid user.  No courses were removed.");
+         } else {
+            try {
+               // switch the current user to the one specified to run the quartz job
+               Session sakaiSesson = sessionManager.getCurrentSession();
+               sakaiSesson.setUserId(user.getId());
+
+               int numSitesRemoved = courseSiteRemovalService.removeCourseSites(action, numDaysAfterTermEnds);
+               logger.info(numSitesRemoved + " course sites were removed.");
+            } catch (Exception ex) {
+               logger.error(ex);
+            }
+         }
+      }
+   }
+
+   /**
+    * @return the enum action corresponding to the string or null if the string is not a valid action.
+    * <br/><br/>
+    * @param string   the action that is to be parsed.
+    */
+   private CourseSiteRemovalService.Action getAction(String string) {
+      CourseSiteRemovalService.Action action = null;
```

Comment:
As Action is an Enum, we need to pass the enumType too and if the string is not one of the valid strings, valueOf in an enum launches an IllegalArgumentException so we need to manage the exceptions and return null in that case... really not saving a lot of lines of code...